spec/console_spec.rb

```diff
@@ -0,0 +1,213 @@
+  let(:win) { '++++' }
+  let(:agree_command) { 'Yes' }
+  let(:disagree_command) { 'No' }
+  let(:name) { 'Rostik' }
+  let(:name_difficulties) { { difficulties: 'easy, medium, hell' } }
+  let(:easy) { 'Easy' }
+  let(:medium) { 'Medium' }
+  let(:hell) { 'Hell' }
+
+  describe '#check_option' do
+    it 'calling method registration' do
+      allow(subject).to receive(:show_info).with(:welcome)
+      allow(subject).to receive(:show_message_with_input).and_return(Console::OPTIONS[:start])
+      expect(subject).to receive(:registration)
+      subject.check_option
+    end
+
+    it 'calling method rules' do
+      allow(subject).to receive(:show_message_with_input).and_return(Console::OPTIONS[:rules], Console::OPTIONS[:start])
+      allow(subject).to receive(:registration)
+      expect(subject).to receive(:show_rules)
+      subject.check_option
+    end
+
+    it 'calling method show_stats' do
+      allow(subject).to receive(:show_info).with(:welcome)
+      allow(subject).to receive(:show_message_with_input).and_return(Console::OPTIONS[:stats], Console::OPTIONS[:start])
+      allow(subject).to receive(:registration)
+      expect(subject).to receive(:show_stats)
+      subject.check_option
+    end
+
+    it 'when invalid' do
+      allow(subject).to receive(:show_info).with(:welcome)
+      allow(subject).to receive(:show_message_with_input).and_return('', Console::OPTIONS[:start])
+      allow(subject).to receive(:registration)
+      expect(subject).to receive(:show_info).with(:wrong_input_option)
+      subject.check_option
+    end
+
+    it 'enter name and difficulty' do
+      allow(subject).to receive(:show_info).with(:welcome)
+      allow(subject).to receive(:show_message_with_input).with(:choose_option).and_return(Console::OPTIONS[:start])
+      expect(subject).to receive(:show_message_with_input).with(:write_name).and_return('')
+      expect(subject).to receive(:show_info).with(:unexpected_command)
+      expect(subject).to receive(:show_message_with_input).with(:write_name).and_return(name)
+      expect(subject).to receive(:show_message_with_input).with(:choose_difficulty, name_difficulties).and_return('')
+      expect(subject).to receive(:show_info).with(:unexpected_command)
+      expect(subject).to receive(:show_message_with_input).with(:choose_difficulty, name_difficulties).and_return(easy)
+      expect(subject).to receive(:play_game)
+      subject.check_option
+    end
+
+    it 'enter difficulty medium' do
+      allow(subject).to receive(:show_info).with(:welcome)
+      allow(subject).to receive(:show_message_with_input).and_return(Console::OPTIONS[:start])
+      allow(subject).to receive(:show_message_with_input).with(:write_name).and_return(name)
+      expect(subject).to receive(:show_message_with_input).with(:choose_difficulty, name_difficulties).and_return(medium)
+      expect(subject).to receive(:play_game)
+      subject.check_option
+    end
+
+    it 'enter difficulty hell' do
```

Comment:
don't check just hell string - you must check everything from your DIFFICULTIES constant